src/runtime_src/core/tools/xbmgmt2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,502 @@
+ * set data retention
+ */
+static void 
+memory_retention(xrt_core::device* device, bool enable)
+{
+  XBU::sudo_or_throw("Updating memory retention requires sudo");
+  auto value = xrt_core::query::data_retention::value_type(enable);
+  xrt_core::device_update<xrt_core::query::data_retention>(device, value);
+}
+
+void
+SubCmdConfigure::execute(const SubCmdOptions& _options) const
+{
+    XBU::verbose("SubCommand: configure");
+    // -- Retrieve and parse the subcommand options -----------------------------
+    // Common options
+    std::vector<std::string> devices;
+    std::string path;
+    std::string retention;
+    bool help = false;
+    // Hidden options
+    bool daemon = false;
+    std::string host;
+    std::string security;
+    std::string clk_scale;
+    std::string power_override;
+    std::string temp_override;
+    std::string cs_reset;
+    bool showx = false;
+
+    // Options previously under the load config command
+    po::options_description loadConfigOptions("Load Config Options");
+    loadConfigOptions.add_options()
+      ("input", boost::program_options::value<decltype(path)>(&path),"INI file with the memory configuration")
+    ;
+
+    // Options previously under the config command
+    po::options_description configOptions("Config Options");
+    configOptions.add_options()
+      ("retention", boost::program_options::value<decltype(retention)>(&retention),"Enables / Disables memory retention.  Valid values are: [ENABLE | DISABLE]")
+    ;
+
+    po::options_description commonOptions("Common Options");
+    commonOptions.add_options()
+        ("device,d", boost::program_options::value<decltype(devices)>(&devices)->multitoken(), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+        ("help,h", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+    ;
+
+    commonOptions.add(loadConfigOptions);
+    commonOptions.add(configOptions);
+
+    // Hidden options previously under the config command
+    po::options_description configHiddenOptions("Hidden Options");
+    configHiddenOptions.add_options()
+        ("daemon", boost::program_options::bool_switch(&daemon), "<add description>")
+        ("host", boost::program_options::value<decltype(host)>(&host), "ip or hostname for peer")
+        ("security", boost::program_options::value<decltype(security)>(&security), "<add description>")
+        ("runtime_clk_scale", boost::program_options::value<decltype(clk_scale)>(&clk_scale), "<add description>")
+        ("cs_threshold_power_override", boost::program_options::value<decltype(power_override)>(&power_override), "<add description>")
+        ("cs_threshold_temp_override", boost::program_options::value<decltype(temp_override)>(&temp_override), "<add description>")
+        ("cs_reset", boost::program_options::value<decltype(cs_reset)>(&cs_reset), "<add description>")
+        ("showx", boost::program_options::bool_switch(&showx), "<add description>")
```

Comment:
I remember Soren changed show to showx because there was some conflict with a global option. Now that we have removed short options for the global options, can you try "show" and see if there is any issue?

Reply:
Oh ok, yeah it mistakes _show_ for _show-hidden_. _showx_ is fine for now